collector.go

```diff
@@ -123,7 +124,30 @@ func getDataHTTP(u *url.URL) ([]byte, error) {
 		Host:       u.Host,
 	}
 
-	resp, err := http.DefaultClient.Do(&req)
+	var client *http.Client
+	if c.exporter.authConfig.ClientCert != "" {
```

Comment:
We could load this when we first start the process and exit the process if there is an issue with loading authentication information initially.